### PR TITLE
Add support for summary metrics

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Add support for summary metrics: a combined count and sum of aggregated measurements, a la Prometheus summaries. We also add partial support for OpenTelemetry summary kind metrics; we store the count and sum, but drop quantiles.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
~- [ ] Documentation has been updated~

## How to test these changes

1. Instrument an application with a Prometheus summary metric
1. Using opentelemetry-collector-contrib, scrape Prometheus endpoint with the [prometheus receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver), and export to APM Server as OTLP
1. Check that summary metric is indexed into `metrics-apm.app.*-default` as an `aggregate_metric_double` field type

## Related issues

Closes https://github.com/elastic/apm-server/issues/5359

### CODE REVIEW

File: processor/otel/metrics.go [hunk 1]
```diff
@@ -371,6 +376,16 @@ func numberSample(dp pdata.NumberDataPoint, metricType model.MetricType) (model.
 	}, true
 }
 
+func summarySample(dp pdata.SummaryDataPoint) (model.MetricsetSample, bool) {
```
Comment (stuartnelson3) [inline processor/otel/metrics.go:379]: is there a reason for the `bool`? seems like that could be dropped, or the struct declaration even inlined.

**Review by stuartnelson3 — APPROVED** [review]
question about a function signature, but that's all

File: processor/otel/metrics.go (hunk 1 above)
Comment (axw) [reply processor/otel/metrics.go:379]: Just being a bit lazy and copy-pasting the code patterns for other metric types. I'll drop the bool.

**Comment by lahsivjar** [comment]
Confirmed successfully on 8.4.0 BC ([45083f33](https://github.com/elastic/apm-server/commits/45083f332df130339dd2174f235409ac5f693469))

<details>
<summary>Prometheus instrumented code</summary>

```golang
package main

import (
	"math/rand"
	"net/http"
	"time"

	"github.com/prometheus/client_golang/prometheus"
	"github.com/prometheus/client_golang/prometheus/promauto"
	"github.com/prometheus/client_golang/prometheus/promhttp"
)

var (
	summaryTest = promauto.NewSummaryVec(
		prometheus.SummaryOpts{
			Name:       "test_summary",
			Help:       "Test summary metrics",
			Objectives: map[float64]float64{0.5: 0.05, 0.99: 0.001},
		},
		[]string{"service_name"},
	)
)

func recordMetrics() {
	go func() {
		for {
			summaryTest.WithLabelValues("test_issue_7772").Observe(rand.Float64())
			time.Sleep(5 * time.Second)
		}
	}()
}

func main() {
	recordMetrics()

	http.Handle("/metrics", promhttp.Handler())
	http.ListenAndServe(":8080", nil)
}
```
</details>

<details>
<summary>Open telemetry collector configuration</summary>

```yaml
receivers:
  prometheus:
    config:
      scrape_configs:
        - job_name: 'otel-collector'
          scrape_interval: 5s
          static_configs:
            - targets: ['localhost:8080']

exporters:
  otlp:
    endpoint: <apm_server_url>
    tls:
      insecure: false
    headers: {"Authorization": "Bearer <secret_token>"}

processors:
  batch:

extensions:
  health_check:

service:
  extensions: [health_check]
  pipelines:
    metrics:
      receivers: [prometheus]
      processors: [batch]
      exporters: [otlp]

```
</details>

<details>
<summary>Documentsuccessfully ingested in ES (snapshot)</summary>

Screenshot 2022-08-02 at 1 58 28 PM

</details>

**Pull request merged**